homeassistant/components/freebox/device_tracker.py

```diff
@@ -1,65 +1,174 @@
 """Support for Freebox devices (Freebox v6 and Freebox mini 4K)."""
-from collections import namedtuple
+from datetime import datetime
 import logging
+from typing import Dict
 
-from homeassistant.components.device_tracker import DeviceScanner
+from homeassistant.components.device_tracker import SOURCE_TYPE_ROUTER
+from homeassistant.components.device_tracker.config_entry import TrackerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.typing import HomeAssistantType
 
-from . import DATA_FREEBOX
+from .const import DEFAULT_DEVICE_NAME, DOMAIN, TRACKER_UPDATE
 
 _LOGGER = logging.getLogger(__name__)
 
 
 async def async_get_scanner(hass, config):
-    """Validate the configuration and return a Freebox scanner."""
-    scanner = FreeboxDeviceScanner(hass.data[DATA_FREEBOX])
-    await scanner.async_connect()
-    return scanner if scanner.success_init else None
-
-
-Device = namedtuple("Device", ["id", "name", "ip"])
-
-
-def _build_device(device_dict):
-    return Device(
-        device_dict["l2ident"]["id"],
-        device_dict["primary_name"],
-        device_dict["l3connectivities"][0]["addr"],
-    )
-
-
-class FreeboxDeviceScanner(DeviceScanner):
-    """Queries the Freebox device."""
-
-    def __init__(self, fbx):
-        """Initialize the scanner."""
-        self.last_results = {}
-        self.success_init = False
-        self.connection = fbx
-
-    async def async_connect(self):
-        """Initialize connection to the router."""
-        # Test the router is accessible.
-        data = await self.connection.lan.get_hosts_list()
-        self.success_init = data is not None
-
-    async def async_scan_devices(self):
-        """Scan for new devices and return a list with found device IDs."""
-        await self.async_update_info()
-        return [device.id for device in self.last_results]
-
-    async def get_device_name(self, device):
-        """Return the name of the given device or None if we don't know."""
-        name = next(
-            (result.name for result in self.last_results if result.id == device), None
+    """Old way of setting up the platform."""
+    pass
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up the device_tracker."""
+    fbx = hass.data[DOMAIN]
+
+    entities = []
+
+    for device in fbx.devices.values():
+        entities.append(device)
+
+    async_add_entities(entities)
+
+
+class FreeboxDevice(TrackerEntity):
```

Comment:
If this device is a router we should use the `ScannerEntity` class as base class. The `TrackerEntity` class is for GPS tracking devices.